stacks-core/release/check-release/action.yml

```
@@ -0,0 +1,80 @@
+## Composite action to check if the branch is a release tag or not
+name: Check Release
+
+inputs:
+  tag:
+    description: "The branch name against which the step runs"
+    required: true
+
+outputs:
+  node_tag:
+    description: "The node release tag, if there is one (empty otherwise)."
+    value: ${{ steps.check_release.outputs.node_tag }}
+  node_docker_tag:
+    description: "The node release tag for docker, if there is one (empty otherwise)."
+    value: ${{ steps.check_release.outputs.node_docker_tag }}
+  signer_tag:
+    description: "The signer release tag, if there is one (empty otherwise)."
+    value: ${{ steps.check_release.outputs.signer_tag }}
+  signer_docker_tag:
+    description: "The signer release tag for docker, if there is one (empty otherwise)."
+    value: ${{ steps.check_release.outputs.signer_docker_tag }}
+  is_node_release:
+    description: "True if the branch is a release one, false otherwise."
+    value: ${{ steps.check_release.outputs.is_node_release }}
+  is_signer_release:
+    description: "True if the branch is a release one, false otherwise."
+    value: ${{ steps.check_release.outputs.is_signer_release }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Set Release Outputs
+      id: check_release
+      shell: bash
+      run: |
+        branch_name=${{ inputs.tag }}
+
+        stacks_core_version_regex="([0-9].[0-9].[0-9].[0-9].[0-9]+(-rc[0-9]+)?)"    # matches x.x.x.x.x || x.x.x.x.x-rcx
+        signer_version_regex="([0-9].[0-9].[0-9].[0-9].[0-9].[0-9]+(-rc[0-9]+)?)"   # matches x.x.x.x.x.x || x.x.x.x.x.x-rcx
```

If in the future we have a version that's stable but we churn out a bunch of MINOR or MAJOR changes over time, this may break. May be safer like this:

```suggestion
        stacks_core_version_regex="(([0-9]+.){4}[0-9]+(-rc[0-9]+)?)"    # matches x.x.x.x.x || x.x.x.x.x-rcx
        signer_version_regex="(([0-9]+.){5}[0-9]+(-rc[0-9]+)?)"   # matches x.x.x.x.x.x || x.x.x.x.x.x-rcx
```

---

can you explain why it may break? unless we change the versioning scheme, i don't see why it would be a problem. 
i do like your suggestion better though, easier to read/update nevertheless.

---

> can you explain why it may break

Sorry not sure how I forgot to explain it. But the original regex would not match against a version like `3.0.0.0.12-rc2` or `3.0.0.14.7-rc2`. While we've yet to have a double digit number in a version placeholder, it's safer and easy to just account for it by adding a `+` in the regex to these digits.